src/cache/dir_cache.go

```diff
@@ -182,12 +180,18 @@ func (cache *dirCache) storeFile(target *core.BuildTarget, out, cacheDir string)
 	return size
 }
 
-func (cache *dirCache) Retrieve(target *core.BuildTarget, key []byte) bool {
-	return cache.retrieveFiles(target, key, "", cacheArtifacts(target))
-}
-
-func (cache *dirCache) RetrieveExtra(target *core.BuildTarget, key []byte, out string) bool {
-	return cache.retrieveFiles(target, key, out, []string{out})
+func (cache *dirCache) Retrieve(target *core.BuildTarget, key []byte) *core.BuildMetadata {
+	outs := target.Outputs()
+	if needsPostBuildFile(target) {
+		outs = append(outs, target.PostBuildOutputFileName())
+	}
+	if !cache.retrieveFiles(target, key, "", outs) {
+		return nil
+	}
+	if needsPostBuildFile(target) {
+		return loadPostBuildFile(target)
+	}
+	return &core.BuildMetadata{}
```

Comment:
This looks weird - why are there no contents?

Reply:
there's no stdout because no post-build, and this cache implementation doesn't store any timestamps